### PR TITLE
EmptyLoopConditionFixer - introduction

### PR DESCRIPTION
**Pull request by SpacePossum**
```
Description of empty_loop_condition rule.
Empty loop-condition must be in configured style.

Fixer is configurable using following option:
* style ('for', 'while'): style of empty loop-condition; defaults to 'while'
```

```diff
--- Original
+++ New
@@ -1,8 +1,8 @@
 <?php
-for(;;) {
+while(true) {
     foo();
 }
 
-do {
+while(true) {
     foo();
-} while(true); // do while
+}  // do while
```

Fixing with configuration: ['style' => 'for']

```diff
--- Original
+++ New
@@ -1,4 +1,4 @@
 <?php
-while(true) {
+for(;;) {
      foo();
 }
```

### CODE REVIEW

File: tests/Fixer/ControlStructure/EmptyLoopConditionFixerTest.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            ['style' => 'for'],
+        ];
+
+        // space cases
+        yield 'lot of space' => [
+            '<?php while(true){ foo3(); }              ',
+            '<?php do{ foo3(); } while(true)            ; ',
+        ];
+
+        yield [
```
Comment (keradus) [inline tests/Fixer/ControlStructure/EmptyLoopConditionFixerTest.php:93]: this change shows that meaning of code is executed.
`while` loop may never execute the body of control statement, while `do-while` executes it at least once

https://3v4l.org/L9ILN#v8.0.9

Comment (SpacePossum) [reply tests/Fixer/ControlStructure/EmptyLoopConditionFixerTest.php:93]: I don't think so, only empty conditions are fixed, not variables, just true

Comment (SpacePossum) [reply tests/Fixer/ControlStructure/EmptyLoopConditionFixerTest.php:93]: Ican add your example as test, showing it won't get touched

Comment (keradus) [reply tests/Fixer/ControlStructure/EmptyLoopConditionFixerTest.php:93]: would love to add it. even if current code is smart enough to handle it, let's avoid someone introducing this issue while refactoring

Comment (SpacePossum) [reply tests/Fixer/ControlStructure/EmptyLoopConditionFixerTest.php:105]: thanks, added :+1:

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**